### PR TITLE
Include the GITHUB_TOKEN/GITEA_TOKEN secret for fork pull requests

### PR DESCRIPTION
**Pull request by js6pak**
Include `GITHUB_TOKEN`/`GITEA_TOKEN` secrets for actions triggered by pull requests

This makes it consistent with the environment variables which you can already access

```shell
echo env: $GITHUB_TOKEN
echo expression: ${{ secrets.GITHUB_TOKEN }}
```
before
image
after
image

### CODE REVIEW

File: routers/api/actions/runner/utils.go [hunk 1]
```diff
@@ -55,6 +55,14 @@ func pickTask(ctx context.Context, runner *actions_model.ActionRunner) (*runnerv
 
 func getSecretsOfTask(ctx context.Context, task *actions_model.ActionTask) map[string]string {
 	secrets := map[string]string{}
+
+	if _, ok := secrets["GITHUB_TOKEN"]; !ok {
+		secrets["GITHUB_TOKEN"] = task.Token
+	}
+	if _, ok := secrets["GITEA_TOKEN"]; !ok {
+		secrets["GITEA_TOKEN"] = task.Token
+	}
```
Comment (wolfogre) [inline routers/api/actions/runner/utils.go]: Maybe
```suggestion
	secrets["GITHUB_TOKEN"] = task.Token
	secrets["GITEA_TOKEN"] = task.Token
```

File: routers/api/actions/runner/utils.go [hunk 2]
```diff
@@ -55,6 +55,14 @@ func pickTask(ctx context.Context, runner *actions_model.ActionRunner) (*runnerv
...
+
 	if task.Job.Run.IsForkPullRequest && task.Job.Run.TriggerEvent != actions_module.GithubEventPullRequestTarget {
 		// ignore secrets for fork pull request
```
Comment (wolfogre) [inline routers/api/actions/runner/utils.go]:
```suggestion
		// ignore secrets for fork pull request, except GITHUB_TOKEN and GITEA_TOKEN which are automatically generated.
```

**Review by wolfogre — APPROVED** [review]

**Comment by wolfogre** [comment]
Please hold on, @CaiCandong will check how GitHub handles this.

**Comment by CaiCandong** [comment]
> Please hold on, @CaiCandong will check how GitHub handles this.

Github  https://github.com/CaiCandong/test/actions/runs/6007638361/job/16294056722
image

**Review by appleboy — APPROVED** [review]

**Comment by wolfogre** [comment]
So it's a bug, and need be backported.

**Review by silverwind — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**